OpenRA.Mods.Common/Activities/Air/Fly.cs

```diff
@@ -88,6 +89,12 @@ public override IEnumerable<Target> GetTargets(Actor self)
 		{
 			yield return target;
 		}
+
+		public override TargetLineNode TargetLineNode(Actor self)
+		{
+			var color = NextActivity == null ? Color.Green : NextActivity.TargetLineNode(self).Color;
```

Comment:
I wish I could do that, it would be much easier. However, even Attack order involves "moving to the target" and makes most part visualization drawing in green. My guess is that if ALL activities can be refactored into parent-child activity and, if all I have to do becomes "traverse NextInQueue" then the color can be statically determined.